### ISSUE TITLE
Define gluster cluster ID

### ISSUE THREAD

**Issue by prashanthpai** [gluster/glusterd2#211]
Currently, there is no separate cluster ID associated with a glusterd2 instance. For all practical purposes, the etcd cluster ID is considered to be glusterd2 cluster ID. However, not all glusterd instances shall be etcd servers in future. We need better definition/mapping of gluster cluster ID and etcd cluster ID. This should allow not so common deployments like - making one beefy server run multiple glusterd2 instances and these instances can belong to different clusters altogether.

**Comment by prashanthpai** [comment]
@kshlm @aravindavk @atinmu @amarts 

I'm planning to take this up. I propose the following approach:

* When glusterd comes up for the first time, it shall generate a cluster ID and be a single node cluster. This is how it is today but there's no cluster ID yet.

* Let's assume we have two independent glusterd instances named `node1` and `node2`. At the beginning, each of these shall have their own unique cluster ID. If `node1` peer probes `node2`, then `node1`'s cluster ID shall prevail as the cluster ID for the newly formed two-node cluster. IOW, peer probe request shall be treated as an invite to join an existing cluster.

* Any node in a fully formed cluster (two or more nodes) can then peer probe any other single glusterd2 instance.

* A single node gluster instance will be disallowed from peer probing a fully formed cluster (two or more nodes) cluster. IOW, joining an existing formed cluster shall be invite only. 

Further....

* Cluster ID shall be part of all inter-node communications (like transaction RPCs). This will prevent cross-cluster communication mismatch.
* External utilities (like heketi and others) can depend on cluster ID to uniquely identify a cluster instead of generating and maintaining their own.
* Etcd namespace shall be segregated (prefixed) by cluster ID. Hence the same etcd store (if external) can be utilised by multiple glusterfs clusters.

No cluster operations based on cluster ID shall be made available by glusterd2 though. Peer attach and detach shall work as before. To the end user, the only noticeable change shall be an additional `ClusterID` field in the peer API responses.

What are your thoughts ?

**Comment by atinmu** [comment]
>What are your thoughts ?

Looks good to me.

**Comment by prashanthpai** [comment]
> To the end user, the only noticeable change shall be an additional ClusterID field in the peer API responses

On a second thought. The Cluster ID field shall be the same for all peers in the same store. Hence not adding it to the peer object itself. The cluster ID shall be sent as a response header for all requests.

**Comment by prashanthpai** [comment]
Sample output:

```sh
curl -i -X GET http://127.0.0.1:24007/v1/peers
HTTP/1.1 200 OK
Content-Type: application/json; charset=UTF-8
X-Gluster-Cluster-Id: 090e8748-8d44-40e7-9942-f62ace3ef524
X-Gluster-Node-Id: b842336d-4347-492c-b54b-efe092caa240
X-Request-Id: 8e3c58cd-b8ba-4cc1-b767-499b2c05bfa5
Date: Thu, 30 Nov 2017 11:27:08 GMT
Content-Length: 109

[{"id":"b842336d-4347-492c-b54b-efe092caa240","name":"gd2-1","addresses":["127.0.0.1:24008"],"online":true}]
```

**Comment by prashanthpai** [comment]
Implemented in PR #463
